File: backend/types/analytics.go
package types

import "time"

// IncidentTrendPoint is a single time-bucket data point returned by the trend query
type IncidentTrendPoint struct {
	TimeBucket string `json:"time_bucket"`
	Status     string `json:"status"`
	Count      int    `json:"count"`
}

// MTTRResult holds mean time to resolve and sla compliance metrics
type MTTRResult struct {
	MTTRMinutes    float64 `json:"mttr_minutes"`
	TotalResolved  int     `json:"total_resolved"`
	SLABreaches    int     `json:"sla_breaches"`
	ComplianceRate float64 `json:"compliance_rate"`
}

// BreachedIncident is an enriched projection for incidents that exceeded the sla target
type BreachedIncident struct {
	ID              string     `json:"id"`
	Title           string     `json:"title"`
	CreatedAt       time.Time  `json:"created_at"`
	ResolvedAt      *time.Time `json:"resolved_at"`
	DurationMinutes float64    `json:"duration_minutes"`
}

File: frontend/src/types/analytics.ts
export interface IncidentTrendPoint {
  time_bucket: string;
  status: string;
  count: number;
}

export interface MTTRResult {
  mttr_minutes: number;
  total_resolved: number;
  sla_breaches: number;
  compliance_rate: number;
}

export interface BreachedIncident {
  id: string;
  title: string;
  created_at: string;
  resolved_at: string | null;
  duration_minutes: number;
}
